astropy/nddata/decorators.py

```diff
@@ -258,7 +248,10 @@ def wrapper(data, *args, **kwargs):
                         AstropyUserWarning,
                     )
 
-            result = func(data, *args, **kwargs)
+            # Need to apply defaults so that any positional only arguments are
+            # passed as positional properly
+            bound_args.apply_defaults()
```

Comment:
wow, this is so cool ! I'm learning a lot in this review !

Reply:
I learnt a lot writing it as well lol